.kiro/specs/auterity-ai-platform-unified/templates/preview-api-specification.py

```
@@ -0,0 +1,748 @@
+    def _generate_cache_key(self, request: PreviewRequest) -> str:
+        """Generate cache key for request"""
+        
+        key_data = {
+            'node_type': request.node_type,
+            'config': request.config,
+            'industry_context': request.industry_context,
+            'compliance_level': request.compliance_level
+        }
+        
+        key_string = json.dumps(key_data, sort_keys=True)
+        return hashlib.md5(key_string.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)